File: DAY-01/find-missing-and-repeating_lx.kt
class Solution {
    fun findMissingAndRepeating(arr: Array<Int>, n: Int) : List<Int> {
        var ans = arrayListOf<Int>()
        for (i in 0 until n) {
            val pos: Int = Math.abs(arr[i]) - 1;
            if (arr[pos] > 0) {
                arr.set(pos, -arr[pos])
            } else {
                ans.add(pos + 1)
            }
        }
        for (i in 0 until n) {
            if (arr[i] > 0) {
                ans.add(i+1)
                break
            }
        }
        return ans
    }
}

File: DAY-01/find-missing-and-repeating_lx.cpp
// problem : https://practice.geeksforgeeks.org/problems/find-missing-and-repeating2512/1#

class Solution{
public:
    int *findTwoElement(int *arr, int n) {
        static int ba[2] = {0, 0};
        for (int i = 0; i < n; i++) {
            if (arr[abs(arr[i]) - 1] > 0) {
                arr[abs(arr[i]) - 1] = - arr[abs(arr[i]) - 1];
            } else {
                ba[0] = abs(arr[i]);
            }
        }
        for (int i = 0; i < n; i++) {
            if (arr[i] > 0) {
                ba[1] = i+1;
            }
        }
        return ba;
    }
};
